src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -296,11 +417,23 @@ public void setNumberOfInstances(int numberOfInstances) {
 				log.error(String.format("retrieving summary for space id '%s' resulted in an exception", spaceIdString), e);
 				return Mono.just(INVALID_SUMMARY);
 			});
+  }
+  
+  private Mono<List<RouteResource>> getAppRoutes(String appId) {
+		return this.cfAccessor.retrieveAppRoutes(appId)
+			.flatMap(response -> {
+				List<RouteResource> routes = response.getResources();
+				if (routes == null) {
+					return Mono.just(INVALID_ROUTES);
```

Comment:
Hard to understand here, as `INVALID_ROUTES` is null

Reply:
I changed this to return an empty list rather than null. I think that makes more sense. 

We can then decide whether to drop that target or not in the flux rather than getting an error.